### PR TITLE
Github report coverage

### PR DESCRIPTION
**Pull request by facuMH**
This PR adds a github workflow, a makefile target and a config file. These combined produce a report on the coverage and (when enabled) post this report on the PR.

This is currently disabled because the unit-test coverage of some packages does not meet the current requirements. This needs to be fixed in a separate PR or lowered thresholds. 

NOTE: 
a series of follow-up PRs will add unit tests to increase coverage and eventually empty the list of overrides

### CODE REVIEW

File: .testcoverage.yml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+# from coverage statistics
+exclude:
+  # Exclude files or packages matching their paths
+  paths:
+    - ^(.+)mock(.+)         # mock files should not be tested
+    - go/ct/common/error.go # error file only declares error type without any logic
+    - go/ct/driver/         # entry points, logic should be tested elsewhere.
+    - go/examples           # examples are not tested.
+    - go/interpreter/lfvm/errors.go # error file only declares error type without any logic
+    - go/interpreter/lfvm/instructions.go # these are tested by driver
```
Comment (simonlechner) [inline .testcoverage.yml:146]: I think eventually there should also be unit tests for instructions in lfvm.

File: .testcoverage.yml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  # The minimum total coverage project should have
+  total: 68
+
+# Holds regexp rules which will override thresholds for matched files or packages 
+# using their paths.
+#
+# First rule from this list that matches file or package is going to apply 
+# new threshold to it. If project has multiple rules that match same path, 
+# override rules should be listed in order from specific to more general rules.
+override:
```
Comment (simonlechner) [inline .testcoverage.yml:44]: Why override all these thresholds and not just lower the threshold for files and packages?

**Review by simonlechner — COMMENTED** [review]
Is it required to specify thresholds and not just get the current coverage? I don't think it makes sense to set thresholds now, while we still have to override them for a substantial amount of the packages.

File: .testcoverage.yml (hunk 2 above)
Comment (facuMH) [reply .testcoverage.yml:44]: I opted for this path because it forces us to list what needs fixing rather than accepting everything.

File: .testcoverage.yml (hunk 1 above)
Comment (facuMH) [reply .testcoverage.yml:146]: like the list of overrides, this can also be an issue with its own pr and tests.

**Comment by facuMH** [comment]
> Is it required to specify thresholds and not just get the current coverage? I don't think it makes sense to set thresholds now, while we still have to override them for a substantial amount of the packages.

it is not required. And I agree, partly, that setting the thresholds now is a little pointless. That said, I still see value in having a value there, and forcing us to list what needs fixing.

**Review by LuisPH3 — APPROVED** [review]

File: go/ct/common/hash.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 	"pgregory.net/rand"
 )
 
-func GetRandomHash(rnd *rand.Rand) tosca.Hash {
+func GetRandomHash(rnd *rand.Rand) tosca.Hash { // coverage-ignore nothing to test
```
Comment (LuisPH3) [inline go/ct/common/hash.go:18]: two things, why do we need this? 
second: I tried it locally, and it does not ignore my code, is it correct?

Comment (facuMH) [reply go/ct/common/hash.go:18]: this comment was the first try to exclude/ignore files/pkg/blocks. Also this file should not be considered for testing since it has only one function that calls to `rnd.Read`, so it would be like testing that function.

Comment (HerbertJordan) [reply go/ct/common/hash.go:18]: I do not think this is a good argument for why this should not be tested. Please remove the `coverage-ignore` comment.

A test just running it to make sure it does not crash would be sufficient in this case.

**Review by HerbertJordan — DISMISSED** [review]

File: go/ct/common/hash.go (hunk 3 above)
Comment (facuMH) [reply go/ct/common/hash.go]: a test has been added for this in the follow-up PR #639

**Review by HerbertJordan — DISMISSED** [review]

**Review by simonlechner — DISMISSED** [review]

**Comment by LuisPH3** [comment]
- [ ] fix main branch issue.

**Review by LuisPH3 — APPROVED** [review]
let see if this fix the main branch bug.

**Review by simonlechner — APPROVED** [review]
